src/Compilers/Core/Portable/SourceGeneration/GeneratorDriver.cs

```diff
@@ -42,16 +42,26 @@ internal GeneratorDriver(ParseOptions parseOptions, ImmutableArray<ISourceGenera
             _state = new GeneratorDriverState(parseOptions, optionsProvider, generators, incrementalGenerators, additionalTexts, ImmutableArray.Create(new GeneratorState[generators.Length]), DriverStateTable.Empty, SyntaxStore.Empty, driverOptions, runtime: TimeSpan.Zero, parseOptionsChanged: true);
         }
 
-        public GeneratorDriver RunGenerators(Compilation compilation, CancellationToken cancellationToken = default)
+        // 4.9 BACKCOMPAT OVERLOAD -- DO NOT TOUCH
+        public GeneratorDriver RunGenerators(Compilation compilation, CancellationToken cancellationToken) => RunGenerators(compilation, generatorFilter: null, cancellationToken);
+
+        /// <summary>
+        /// Run generators and produce an updated <see cref="GeneratorDriver"/> containing the results.
+        /// </summary>
+        /// <param name="compilation">The compilation to run generators against</param>
+        /// <param name="generatorFilter">An optional filter that specifies which generators to run. If <c>null</c> all generators will run.</param>
+        /// <param name="cancellationToken">Used to cancel an in progress operation.</param>
+        /// <returns>An updated driver that contains the results of the generators running.</returns>
+        public GeneratorDriver RunGenerators(Compilation compilation, Func<GeneratorFilterContext, bool>? generatorFilter = null, CancellationToken cancellationToken = default)
```

Comment:
I couldn't figure out a way to make `generatorFilter` non-optional (as discussed in the API review) without breaking our back compat guidelines.

Another option would be to just make this a differently named method, like `RunGeneratorsWithFilter`

Reply:
Hmm, I thought it was complaining about it, but it's not. Fixed.

Reply:
Our analyzer requires that we don't have multiple overloads with optional parameters: https://github.com/dotnet/roslyn/blob/main/docs/Adding%20Optional%20Parameters%20in%20Public%20API.md

That means that the signatures when we make `generatorFilter` required are:
```
RunGenerators(Compilation, CancellationToken)
RunGenerators(Compilation, Func<> filter, CancellationToken = default)
```

Anyone who was calling `RunGenerators(Compilation)` will now be source broken, because we no longer have an overload with only a single required parameter.

Note: It *also* says that we shouldn't add new optional parameters in the middle of a signature for source-compat. Imagine if you had `RunGenerator(compilation, default)` that might bind to a different overload now. So we're in a bit of tricky situation.

We can either: make the filter optional, and bend the 'don't put optional in the middle rule', or we can make it non-optional, and introduce a third overload of `RunGenerators` that *only* takes a `Compilation` parameter. 

Tagging @333fred for any API shape thoughts on this one?

Reply:
What is the problem with the signature in the current form? Yes it's optional but it just gets filled in as essentially "run everything". That behavior is compatible with the existing overload so that all seems to track  for me. What am I missing?

Reply:
*shrug* I'm just following the guidelines here https://github.com/dotnet/roslyn/blob/main/docs/Adding%20Optional%20Parameters%20in%20Public%20API.md where it says do not put optional params in the middle of signatures. 

I don't see in this case why it would be an issue, as even in source `RunGenerators(compilation, default)` should still bind to the cancellation token version, but I assume the guidelines exist for a reason and don't want to break them :)